### PR TITLE
monitoring: add docker compose dev env for grafana dashboards

### PR DESCRIPTION
**Pull request by kzh**
This PR adds in a docker compose file used to bootstrap a three node
cockroachdb cluster hooked onto a prometheus instance and a grafana instance.
This is intended to be used for spinning up a quick dev environment for
performing manual QA on the grafana dashboards.

This PR also tweaks the existing grafana dashboards such that they
are compatible with dashboard provisioning in grafana.

See:
grafana/grafana#10786 (comment)

Release note: None

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
Last commit looks good, I was able to test locally. Have a few questions inline for additions.

For this PR can you remove the prior commits in the history so it's easy to merge? There's no reason to wait for review of the dashboards PR to merge this one. You can do this by running `git rebase --interactive master` on your feature branch and removing the commits you don't need from the list.

> Reviewed 11 of 11 files at r4, 5 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-:-Mb8mhLO91i2MyeEbbhI:b3mb8bu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kzh)

___
*[monitoring/README.md, line 31 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8mUTDDdw9vu-1WUON:-Mb8mUTDDdw9vu-1WUOO:bqmd89s) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L31)):*
> ```Markdown
> - `localhost:8080`: cockroachdb console
> - `localhost:9090`: prometheus web ui
> - `localhost:3000`: grafana
> ```

Some additional advice on where to see the loaded dashboards would be helpful here. I'm not super familiar with the Grafana UI and it took me some clicking to find what to look for.
___
*[monitoring/README.md, line 36 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8mB9n-2I7l7M7FswQ:-Mb8mB9o113lLrSDweBZ:bk1kxxk) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L36)):*
> ```Markdown
> 
> Lastly, run this command to init the cluster.  
> `$ cockroach init --insecure`
> ```

Any chance this can be run automatically during provisioning by docker compose?
___
*[monitoring/README.md, line 42 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8nJYY5Jt1F4jGRaQW:-Mb8nJYY5Jt1F4jGRaQX:b-8ocayc) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L42)):*
> ```Markdown
> The `grafana-dashboards` directory is mounted onto the grafana container. Grafana will poll this directory every five
> seconds to ensure dashboards are up to date. After each dashboard modification, the dashboard should be exported as a JSON file
> into this directory.
> ```

This is a bit confusing...I'm unclear on whether changes in the Grafana UI will be persisted for the lifetime of the docker images or not. It seems like Grafana will just reload the configs from disk every 5 seconds...is that true?

Try clarifying the flow step by step of how to make modifications. Should I export and save into the same directory as the existing templates?

**Review by kzh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65964)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[monitoring/README.md, line 31 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8mUTDDdw9vu-1WUON:-MbDRy2b6syowBGkDEOE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L31)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Some additional advice on where to see the loaded dashboards would be helpful here. I'm not super familiar with the Grafana UI and it took me some clicking to find what to look for.
</blockquote></details>

Done.
___
*[monitoring/README.md, line 36 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8mB9n-2I7l7M7FswQ:-MbDRze-8GtxfvroeZq1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L36)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Any chance this can be run automatically during provisioning by docker compose?
</blockquote></details>

Done.
___
*[monitoring/README.md, line 42 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-Mb8nJYY5Jt1F4jGRaQW:-MbDS0jIDFwkuSRylzOR:b-xzkgf7) ([raw file](https://github.com/cockroachdb/cockroach/blob/4914769fe1242d7a79b11e5538eb774391ebeaa7/monitoring/README.md#L42)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

This is a bit confusing...I'm unclear on whether changes in the Grafana UI will be persisted for the lifetime of the docker images or not. It seems like Grafana will just reload the configs from disk every 5 seconds...is that true?

Try clarifying the flow step by step of how to make modifications. Should I export and save into the same directory as the existing templates?
</blockquote></details>

I decided to omit this extraneous detail about the mounting and provided a step by step guide for persisting changes made to dashboards.

**Review by dhartunian — APPROVED** [review]
:lgtm: just one nit before merge: can you add explanation in the commit/PR message for the changes made to the dashboard JSON files?

> Reviewed 14 of 15 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65964#-:-MbIGBN-7EDGXmpFEenu:b-ljg2x3)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian)

**Pull request merged**
